core/resharding.go

```diff
@@ -67,9 +70,12 @@ func (ss *ShardingState) cuckooResharding(percent float64) {
 		}
 		numKicked := int(percent * float64(len(ss.shardState[i].NodeList)))
 		if numKicked == 0 {
-			numKicked++
+			numKicked++ // At least kick one node out
 		}
 		length := len(ss.shardState[i].NodeList)
+		if length-numKicked <= 0 {
```

Comment:
In this case, we can kick out part of numKicked for example first 10%

Reply:
It shouldn't matter if we kick out a constant of 10%.  This condition only happen when numKicked++ happens, and is only to make sure we don't kick out all nodes.